tfe/resource_tfe_workspace_run.go

```diff
@@ -18,42 +18,61 @@ var (
 )
 
 var planPendingStatuses = map[tfe.RunStatus]bool{
-	tfe.RunPending:        true,
-	tfe.RunPlanQueued:     true,
-	tfe.RunPlanning:       true,
-	tfe.RunCostEstimating: true,
-	tfe.RunPolicyChecking: true,
-	tfe.RunQueuing:        true,
-	tfe.RunFetching:       true,
-}
-
-var planTerminalStatuses = map[tfe.RunStatus]bool{
-	tfe.RunPlanned:            true,
-	tfe.RunPlannedAndFinished: true,
-	tfe.RunErrored:            true,
-	tfe.RunCostEstimated:      true,
-	tfe.RunPolicyChecked:      true,
-	tfe.RunPolicySoftFailed:   true,
-	tfe.RunPolicyOverride:     true,
+	tfe.RunPending:           true,
+	tfe.RunPlanQueued:        true,
+	tfe.RunPlanning:          true,
+	tfe.RunCostEstimating:    true,
+	tfe.RunPolicyChecking:    true,
+	tfe.RunQueuing:           true,
+	tfe.RunFetching:          true,
+	tfe.RunPostPlanRunning:   true,
+	tfe.RunPostPlanCompleted: true,
+	tfe.RunPrePlanRunning:    true,
```

Comment:
With the exception of `RunPostPlanCompleted`, I don't understand how some of these can be terminal statuses, when they appear to be transition states... Can you elaborate?